src/main/java/com/dre/brewery/configuration/AbstractConfigManager.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * BreweryX Bukkit-Plugin for an alternate brewing process
+ * Copyright (C) 2024 The Brewery Team
+ *
+ * This file is part of BreweryX.
+ *
+ * BreweryX is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BreweryX is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BreweryX. If not, see <http://www.gnu.org/licenses/gpl-3.0.html>.
+ */
+
+package com.dre.brewery.configuration;
+
+import com.dre.brewery.BreweryPlugin;
+import com.dre.brewery.configuration.annotation.OkaeriConfigFileOptions;
+import com.dre.brewery.configuration.configurer.BreweryXConfigurer;
+import com.dre.brewery.configuration.configurer.TranslationManager;
+import com.dre.brewery.utility.Logging;
+import eu.okaeri.configs.configurer.Configurer;
+import eu.okaeri.configs.serdes.OkaeriSerdesPack;
+import eu.okaeri.configs.serdes.standard.StandardSerdes;
+import eu.okaeri.configs.yaml.snakeyaml.YamlSnakeYamlConfigurer;
+import lombok.Getter;
+import lombok.Setter;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@Getter
+@Setter
+public class AbstractConfigManager {
```

Comment:
I don't like how it's named **Abstract**ConfigManager

Reply:
I agree, I'll change the name but if you have any name suggestions I'm happy to go with that